cmd/pebble/cmd_ls.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	pathpkg "path"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/canonical/pebble/client"
+	"github.com/canonical/pebble/internal/strutil/quantity"
+)
+
+// ErrPatternOutsideFileName is returned if a pattern is found anywhere but the file name of a path
+var ErrPatternOutsideFileName = errors.New("patterns can only be applied to file names, not directories")
+
+type cmdLs struct {
+	clientMixin
+	timeMixin
+
+	Directory  bool `short:"d" long:"directory"`
```

Comment:
Per discussion with Gustavo: let's just use the shorthand notation `-d`, not support `--directory`.

Rationale: over time we've learned that it doesn't help to offer ambiguity on such options. If it is something that's unclear or we only use it rarely, we use the long form so people understand what it is. If you use it all the time then a short notation is good, and people only ever use the short `-d` form. It's quite rare to have an option that's both unclear and used all the time.